### PR TITLE
Refactor ServiceConfig code to pave the way for adding a ConfigSelector API

### PR DESCRIPTION
**Pull request by markdroth**

### CODE REVIEW

File: src/core/ext/filters/client_channel/service_config_call_data.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
+//
+// Copyright 2016 gRPC authors.
```
[the remark this answers is not in the document]
Comment (markdroth) [reply src/core/ext/filters/client_channel/service_config_call_data.h:2]: This code was moved from an existing file, so I kept the date from that file.

File: src/core/ext/filters/client_channel/service_config_parser.cc [hunk 2]
```diff
@@ -0,0 +1,87 @@
+//
+// Copyright 2015 gRPC authors.
```
[the remark this answers is not in the document]
Comment (markdroth) [reply src/core/ext/filters/client_channel/service_config_parser.cc:2]: This code was moved from an existing file, so I kept the date from that file.

File: src/core/ext/filters/client_channel/service_config_parser.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
+//
+// Copyright 2016 gRPC authors.
```
[the remark this answers is not in the document]
Comment (markdroth) [reply src/core/ext/filters/client_channel/service_config_parser.h:2]: This code was moved from an existing file, so I kept the date from that file.

**Comment by yashykt** [comment]
Thanks for doing this

**Comment by markdroth** [comment]
Known issues: #19834

**Pull request merged**